app/addons/documents/components/results-options.js

```diff
@@ -0,0 +1,96 @@
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import Components from '../../components/react-components';
+import Constants from '../constants';
+
+const { MenuDropDown } = Components;
+
+export default class ResultsOptions extends React.Component {
+  static defaultProps = {
+    showDensity: true,
+    showFontSize: true
+  };
+  static propTypes = {
+    resultsStyle: PropTypes.object.isRequired,
+    updateStyle: PropTypes.func.isRequired,
+    showDensity: PropTypes.bool,
+    showFontSize: PropTypes.bool
+  };
+
+  constructor(props) {
+    super(props);
+    this.toggleTextOverflow = this.toggleTextOverflow.bind(this);
+    this.setFontSize = this.setFontSize.bind(this);
+  }
+
+  toggleTextOverflow() {
+    if (this.props.resultsStyle.textOverflow === Constants.INDEX_RESULTS_STYLE.TEXT_OVERFLOW_FULL) {
+      this.props.updateStyle({
+        textOverflow: Constants.INDEX_RESULTS_STYLE.TEXT_OVERFLOW_TRUNCATED
+      });
+    } else {
+      this.props.updateStyle({
+        textOverflow: Constants.INDEX_RESULTS_STYLE.TEXT_OVERFLOW_FULL
+      });
+    }
+  }
+
+  setFontSize(size) {
+    this.props.updateStyle({
+      fontSize: size
+    });
+  }
+
+  getDensitySection() {
+    const densityItems = [{
+      title: 'Show full values',
+      onClick: this.toggleTextOverflow
+    }];
+    if (this.props.resultsStyle.textOverflow === Constants.INDEX_RESULTS_STYLE.TEXT_OVERFLOW_FULL) {
```

Comment:
I think you should define the title before creating the densityItem. It's more error prone if you mutate the array at a certain index after its creation.

Reply:
Agree... it's changed.